domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/searchaliasfield/SearchAliasFieldUnicityValidator.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.searchaliasfield;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.commons.lang3.StringUtils.defaultString;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.annotation.RequestScope;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.SearchAliasFieldEntity;
+
+@Component
+@RequestScope
```

Comment:
how come the RequestScope here?

Reply:
It validates each field at a time, so building a temporary map to catch any duplicate search alias fields per case type for the request.